File: test_shared/src/main/java/com/clevertap/android/shared/test/TestApplication.java
package com.clevertap.android.shared.test;

import androidx.test.core.app.ApplicationProvider;

import com.clevertap.android.sdk.ActivityLifecycleCallback;
import com.clevertap.android.sdk.Application;
import com.clevertap.android.sdk.CleverTapAPI;

public class TestApplication extends Application {

    @Override
    public void onCreate() {
        CleverTapAPI.setUIEditorConnectionEnabled(true);
        CleverTapAPI.setDebugLevel(3);
        ActivityLifecycleCallback.register(this);
        super.onCreate();
    }

    public static TestApplication getApplication() {
        return (TestApplication) ApplicationProvider.getApplicationContext();
    }
}

File: test_shared/src/main/java/com/clevertap/android/shared/test/TestApplication.kt
package com.clevertap.android.shared.test

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import com.clevertap.android.sdk.ActivityLifecycleCallback
import com.clevertap.android.sdk.Application
import com.clevertap.android.sdk.CleverTapAPI

class TestApplication : Application() {

    override fun onCreate() {
        CleverTapAPI.setUIEditorConnectionEnabled(true)
        CleverTapAPI.setDebugLevel(3)
        ActivityLifecycleCallback.register(this)
        super.onCreate()
    }

    companion object {

        val application: TestApplication
            get() = ApplicationProvider.getApplicationContext<Context>() as TestApplication
    }
}